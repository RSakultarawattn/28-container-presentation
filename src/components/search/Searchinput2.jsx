import React from 'react';
import PropTypes from 'prop-types';
//import getArticles from '../../services/newsApi';


function Searchinput2({ onSubmit, search, handleChange }) {
  return (
    <form onSubmit = {onSubmit}>
      <input type={search} placeholder="type words here" 
      onChange={ handleChange }/>

      <button type="submit">

                submit
      </button>

    </form>
  );
}

Searchinput2.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  search: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired  

};

export default Searchinput2;

