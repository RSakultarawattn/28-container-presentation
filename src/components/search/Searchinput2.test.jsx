import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Searchinput2 from './Searchinput2';

const search = '';
function handleChange(){

}
function handleSearch() {

}

describe('search input', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(<Searchinput2 search={search} 
      handleChange={handleChange} 
      onSubmit={handleSearch} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
