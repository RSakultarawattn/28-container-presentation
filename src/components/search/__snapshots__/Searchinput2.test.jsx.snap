// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search input renders App 1`] = `
<DocumentFragment>
  <form>
    <input
      placeholder="type words here"
      type=""
    />
    <button
      type="submit"
    >
      submit
    </button>
  </form>
</DocumentFragment>
`;
